lib/ansible/plugins/inventory/foreman.py

```diff
@@ -231,6 +246,30 @@ def _populate(self):
                 if self.get_option('want_facts'):
                     self.inventory.set_variable(host_name, 'foreman_facts', self._get_facts(host))
 
+                # create group for host collections
+                if self.get_option('want_hostcollections'):
+                    host_data = self._get_host_data_by_id(host['id'])
+                    hostcollections = host_data.get('host_collections')
+                    if hostcollections:
+                        # Create Ansible groups for host collections
+                        for hostcollection in hostcollections:
+                            try:
+                                hostcollection_group = to_safe_group_name('%shostcollection_%s' % (self.get_option('group_prefix'),
+                                                                          hostcollection['name'].lower().replace(" ", "")))
+                                hostcollection_group = self.inventory.add_group(hostcollection_group)
+                                self.inventory.add_child(hostcollection_group, host_name)
+                            except ValueError as e:
+                                self.display.warning("Could not create groups for host collections for %s, skipping: %s" % (host_name, to_text(e)))
+
+                # put ansible_ssh_host as hostvar
+                if self.get_option('want_ansible_ssh_host'):
+                    if host.get('ip'):
+                        try:
+                            self.inventory.set_variable(host_name, 'ansible_ssh_host', host['ip'])
+                        except ValueError as e:
+                            self.display.warning("Could not set hostvar ansible_ssh_host to '%s' for the '%s' host, skipping: %s" %
+                                                 (host['ip'], host_name, to_native(e)))
```

Comment:
Why is this using `to_native`? On line 262 `to_text` is used and I'm not entirely sure which is the correct one.

Reply:
@ekohl I can make it to_text, it was just ignorance on my part.
to_native on python3 is equivalent to to_text. Therefore I will make it to_text.